File: src/Application Specific/Classes/Entity Collections/Customer/CustomerServiceProcess.cs
using System;
using System.Collections;

namespace FSM.Entity.Customers
{
    public class CustomerServiceProcess
    {
        private DataTypeValidator _dataTypeValidator;

        public CustomerServiceProcess()
        {
            _dataTypeValidator = new DataTypeValidator();
        }

        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        public bool IgnoreExceptionsOnSetMethods
        {
            get
            {
                return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
            }

            set
            {
                _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
            }
        }

        public Hashtable Errors
        {
            get
            {
                return _dataTypeValidator.Errors;
            }
        }

        public DataTypeValidator DTValidator
        {
            get
            {
                return _dataTypeValidator;
            }
        }

        private bool _exists = false;

        public bool Exists
        {
            get
            {
                return _exists;
            }

            set
            {
                _exists = value;
            }
        }

        private bool _deleted = false;

        public bool Deleted
        {
            get
            {
                return _deleted;
            }
        }

        public bool SetDeleted
        {
            set
            {
                _deleted = value;
            }
        }

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        /* TODO ERROR: Skipped RegionDirectiveTrivia */
        private int _CustomerServiceProcessID = 0;

        public int CustomerServiceProcessID
        {
            get
            {
                return _CustomerServiceProcessID;
            }
        }

        public object SetCustomerServiceProcessID
        {
            set
            {
                _dataTypeValidator.SetValue(this, "_CustomerServiceProcessID", value);
            }
        }

        private int _CustomerID = 0;

        public int CustomerID
        {
            get
            {
                return _CustomerID;
            }
        }

        public object SetCustomerID
        {
            set
            {
                _dataTypeValidator.SetValue(this, "_CustomerID", value);
            }
        }

        private int _Letter1 = 0;

        public int Letter1
        {
            get
            {
                return _Letter1;
            }
        }

        public object SetLetter1
        {
            set
            {
                _dataTypeValidator.SetValue(this, "_Letter1", value);
            }
        }

        private int _Letter2 = 0;

        public int Letter2
        {
            get
            {
                return _Letter2;
            }
        }

        public object SetLetter2
        {
            set
            {
                _dataTypeValidator.SetValue(this, "_Letter2", value);
            }
        }

        private int _Letter3 = 0;

        public int Letter3
        {
            get
            {
                return _Letter3;
            }
        }

        public object SetLetter3
        {
            set
            {
                _dataTypeValidator.SetValue(this, "_Letter3", value);
            }
        }

        private int _Appointment1 = 0;

        public int Appointment1
        {
            get
            {
                return _Appointment1;
            }
        }

        public object SetAppointment1
        {
            set
            {
                _dataTypeValidator.SetValue(this, "_Appointment1", value);
            }
        }

        private int _Appointment2 = 0;

        public int Appointment2
        {
            get
            {
                return _Appointment2;
            }
        }

        public object SetAppointment2
        {
            set
            {
                _dataTypeValidator.SetValue(this, "_Appointment2", value);
            }
        }

        private int _Appointment3 = 0;

        public int Appointment3
        {
            get
            {
                return _Appointment3;
            }
        }

        public object SetAppointment3
        {
            set
            {
                _dataTypeValidator.SetValue(this, "_Appointment3", value);
            }
        }

        private DateTime _DateAdded = default;

        public DateTime DateAdded
        {
            get
            {
                return _DateAdded;
            }

            set
            {
                _DateAdded = value;
            }
        }

        private int _AddedBy = 0;

        public int AddedBy
        {
            get
            {
                return _AddedBy;
            }
        }

        public object SetAddedBy
        {
            set
            {
                _dataTypeValidator.SetValue(this, "_AddedBy", value);
            }
        }

        private byte[] _template1;

        public byte[] Template1
        {
            get
            {
                return _template1;
            }

            set
            {
                _template1 = value;
            }
        }

        private byte[] _template2;

        public byte[] Template2
        {
            get
            {
                return _template2;
            }

            set
            {
                _template2 = value;
            }
        }

        private byte[] _template3;

        public byte[] Template3
        {
            get
            {
                return _template3;
            }

            set
            {
                _template3 = value;
            }
        }

        private byte[] _patchCheckTemplate;

        public byte[] PatchCheckTemplate
        {
            get
            {
                return _patchCheckTemplate;
            }

            set
            {
                _patchCheckTemplate = value;
            }
        }

        /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
    }
}

File: src/Application Specific/Classes/Entity Collections/Customer/CustomerServiceProcess.vb
Namespace Entity.Customers

    Public Class CustomerServiceProcess

        Private _dataTypeValidator As DataTypeValidator

        Public Sub New()
            _dataTypeValidator = New DataTypeValidator
        End Sub

#Region "Class Properties"

        Public Property IgnoreExceptionsOnSetMethods() As Boolean
            Get
                Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
            End Get
            Set(ByVal Value As Boolean)
                Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
            End Set
        End Property

        Public ReadOnly Property Errors() As Hashtable
            Get
                Return _dataTypeValidator.Errors
            End Get
        End Property

        Public ReadOnly Property DTValidator() As DataTypeValidator
            Get
                Return _dataTypeValidator
            End Get
        End Property

        Private _exists As Boolean = False

        Public Property Exists() As Boolean
            Get
                Return _exists
            End Get
            Set(ByVal Value As Boolean)
                _exists = Value
            End Set
        End Property

        Private _deleted As Boolean = False

        Public ReadOnly Property Deleted() As Boolean
            Get
                Return _deleted
            End Get
        End Property

        Public WriteOnly Property SetDeleted() As Boolean
            Set(ByVal Value As Boolean)
                _deleted = Value
            End Set
        End Property

#End Region

#Region "Customer Properties"

        Private _CustomerServiceProcessID As Integer = 0

        Public ReadOnly Property CustomerServiceProcessID() As Integer
            Get
                Return _CustomerServiceProcessID
            End Get
        End Property

        Public WriteOnly Property SetCustomerServiceProcessID() As Object
            Set(ByVal Value As Object)
                _dataTypeValidator.SetValue(Me, "_CustomerServiceProcessID", Value)
            End Set
        End Property

        Private _CustomerID As Integer = 0

        Public ReadOnly Property CustomerID() As Integer
            Get
                Return _CustomerID
            End Get
        End Property

        Public WriteOnly Property SetCustomerID() As Object
            Set(ByVal Value As Object)
                _dataTypeValidator.SetValue(Me, "_CustomerID", Value)
            End Set
        End Property

        Private _Letter1 As Integer = 0

        Public ReadOnly Property Letter1() As Integer
            Get
                Return _Letter1
            End Get
        End Property

        Public WriteOnly Property SetLetter1() As Object
            Set(ByVal Value As Object)
                _dataTypeValidator.SetValue(Me, "_Letter1", Value)
            End Set
        End Property

        Private _Letter2 As Integer = 0

        Public ReadOnly Property Letter2() As Integer
            Get
                Return _Letter2
            End Get
        End Property

        Public WriteOnly Property SetLetter2() As Object
            Set(ByVal Value As Object)
                _dataTypeValidator.SetValue(Me, "_Letter2", Value)
            End Set
        End Property

        Private _Letter3 As Integer = 0

        Public ReadOnly Property Letter3() As Integer
            Get
                Return _Letter3
            End Get
        End Property

        Public WriteOnly Property SetLetter3() As Object
            Set(ByVal Value As Object)
                _dataTypeValidator.SetValue(Me, "_Letter3", Value)
            End Set
        End Property

        Private _Appointment1 As Integer = 0

        Public ReadOnly Property Appointment1() As Integer
            Get
                Return _Appointment1
            End Get
        End Property

        Public WriteOnly Property SetAppointment1() As Object
            Set(ByVal Value As Object)
                _dataTypeValidator.SetValue(Me, "_Appointment1", Value)
            End Set
        End Property

        Private _Appointment2 As Integer = 0

        Public ReadOnly Property Appointment2() As Integer
            Get
                Return _Appointment2
            End Get
        End Property

        Public WriteOnly Property SetAppointment2() As Object
            Set(ByVal Value As Object)
                _dataTypeValidator.SetValue(Me, "_Appointment2", Value)
            End Set
        End Property

        Private _Appointment3 As Integer = 0

        Public ReadOnly Property Appointment3() As Integer
            Get
                Return _Appointment3
            End Get
        End Property

        Public WriteOnly Property SetAppointment3() As Object
            Set(ByVal Value As Object)
                _dataTypeValidator.SetValue(Me, "_Appointment3", Value)
            End Set
        End Property

        Private _DateAdded As DateTime = Nothing

        Public Property DateAdded() As DateTime
            Get
                Return _DateAdded
            End Get
            Set(ByVal Value As DateTime)
                _DateAdded = Value
            End Set
        End Property

        Private _AddedBy As Integer = 0

        Public ReadOnly Property AddedBy() As Integer
            Get
                Return _AddedBy
            End Get
        End Property

        Public WriteOnly Property SetAddedBy() As Object
            Set(ByVal Value As Object)
                _dataTypeValidator.SetValue(Me, "_AddedBy", Value)
            End Set
        End Property

        Private _template1 As Byte()

        Public Property Template1() As Byte()
            Get
                Return _template1
            End Get
            Set(ByVal value As Byte())
                _template1 = value
            End Set
        End Property

        Private _template2 As Byte()

        Public Property Template2() As Byte()
            Get
                Return _template2
            End Get
            Set(ByVal value As Byte())
                _template2 = value
            End Set
        End Property

        Private _template3 As Byte()

        Public Property Template3() As Byte()
            Get
                Return _template3
            End Get
            Set(ByVal value As Byte())
                _template3 = value
            End Set
        End Property

        Private _patchCheckTemplate As Byte()

        Public Property PatchCheckTemplate() As Byte()
            Get
                Return _patchCheckTemplate
            End Get
            Set(ByVal value As Byte())
                _patchCheckTemplate = value
            End Set
        End Property

#End Region

    End Class

End Namespace